src/middlewares/error_handlers.cr

```diff
@@ -1,29 +1,59 @@
-# :nodoc:
-module ErrorHandlerHelpers
-  extend self
+require "http"
+require "json"
 
-  def messages_from(error : Exception)
-    messages = [response_data_for error]
-    while error = error.cause
-      messages << response_data_for error
-    end
-    messages
+struct ErrorData
+  include JSON::Serializable
+  property type : String
+  property message : String?
+  property status_code : HTTP::Status?
+
+  def initialize(@type, @message, @status_code = nil)
   end
 
-  def response_data_for(error : Exception)
+  def self.from(error : Exception)
```

Comment:
`from_exception` would be a better name, to differentiate with `from_json`

Reply:
With JSON, one usually has an IO or file or something, so one calls `ErrorResponse.from_json(context.response)` or `ErrorResponse.from_json file`. `#from_exception` would stutter with a variable named `exception`, like in the block where it's called from:

```crystal
response_data = ErrorResponse.from_exception exception
```

I *really* don't like that. Can we compromise on a `self.new` overload?

Reply:
Good idea yes